scripts/irods/test/test_ichksum.py

```diff
@@ -168,3 +168,112 @@ def test_chksum_catalog_verify_not_exhausting_statements__issue_4732(self):
+        self.admin.assert_icommand(['ichksum', '-a', '-n', '0', data_object], 'STDERR', ["the 'n' and 'a' option cannot be used together"])
+        self.admin.assert_icommand(['ichksum', '-a', '-R', 'demoResc', data_object], 'STDERR', ["the 'R' and 'a' option cannot be used together"])
+        self.admin.assert_icommand(['ichksum', '-K', '-f', data_object], 'STDERR', ["the 'K' and 'f' option cannot be used together"])
+        self.admin.assert_icommand(['ichksum', '-K', '-a', '-n', '0', data_object], 'STDERR', ["the 'n' and 'a' option cannot be used together"])
+        self.admin.assert_icommand(['ichksum', '-K', '-a', '-R', 'demoResc', data_object], 'STDERR', ["the 'R' and 'a' option cannot be used together"])
+        self.admin.assert_icommand(['ichksum', '-K', '--silent', data_object], 'STDERR', ["the 'K' and 'silent' option cannot be used together"])
+        self.admin.assert_icommand(['ichksum', '--verify', '--silent', data_object], 'STDERR', ["the 'verify' and 'silent' option cannot be used together"])
+
+    def test_ichksum_reports_when_replicas_are_missing_checksums__issue_5252(self):
+        data_object = 'foo'
+        self.admin.assert_icommand(['istream', 'write', data_object], input='some data')
+        self.admin.assert_icommand(['irepl', '-R', self.testresc, data_object])
+        self.admin.assert_icommand(['ichksum', '-n0', data_object], 'STDOUT', [data_object + '    sha2:'])
+        self.admin.assert_icommand(['ichksum', '-K', data_object], 'STDOUT', ['WARNING: No checksum available for replica [1].'])
+
+        # Show that --verify is an alias for -K.
+        self.admin.assert_icommand(['ichksum', '--verify', data_object], 'STDOUT', ['WARNING: No checksum available for replica [1].'])
+
+    def test_ichksum_reports_when_replicas_physical_size_does_not_match_size_in_catalog__issue_5252(self):
+        data_object = os.path.join(self.admin.session_collection, 'foo')
+        self.admin.assert_icommand(['istream', 'write', data_object], input='some data')
+        self.admin.assert_icommand(['iadmin', 'modrepl', 'logical_path', data_object, 'replica_number', '0', 'DATA_SIZE', '1'])
+        self.admin.assert_icommand(['ichksum', '-K', data_object], 'STDOUT', ['ERROR: Physical size does not match size in catalog for replica [0].'])
+
+        # Show that --verify is an alias for -K.
+        self.admin.assert_icommand(['ichksum', '--verify', data_object], 'STDOUT', ['ERROR: Physical size does not match size in catalog for replica [0].'])
+
+    def test_ichksum_reports_when_replicas_do_not_share_identical_checksums__issue_5252(self):
+        data_object = 'foo'
+        self.admin.assert_icommand(['istream', 'write', data_object], input='some data')
+        self.admin.assert_icommand(['irepl', '-R', self.testresc, data_object])
+        self.admin.assert_icommand(['istream', 'write', '-n0', '-o0', '--no-trunc', data_object], input='XXXX')
```

Comment:
these tests need to be revisited. if multiple replicas exist and istream is used to append/overwrite data in one of the replicas, should the sibling replicas be marked stale. if yes, then some of these tests are incorrect because they assume istream will not cause replicas to be marked stale (which is temporarily the case right now).

Reply:
Yes, sibling replicas should be marked stale by iRODS today, and the same when @alanking lands his intermediate/logical_locking code.   The tests should be changed/updated.